docs/pages/repo/docs/troubleshooting.mdx

```diff
@@ -5,37 +5,152 @@ description: This guide aims to help you debug issues with your Turborepo builds
+
+You can [get started][7] with Turborepo with minimal configuration -- that's one
+of the things people love about Turborepo! But when you omit configuration,
+Turborepo internally falls back to smart defaults. Additionally, when using
+[Workspace Configurations][d-config-workspaces] in a monorepo, it can be
+confusing to understand how Turborepo interpreted your `turbo.json`. You can use
+the `--dry` or `--dry=json` to get a "resolved" task configuration for any task.
+For example:
+
+```bash
+turbo run build --dry=json
+```
+
+Look for a `resolvedTaskConfiguration` key in the output.
+
+### User Config
+
+When you link your repository to Vercel, Turborepo stores configuration in two places:
+
+- your Vercel team information is stored in `.turbo/config.json`. You can
+  inspect this file to see what else might be in there!
+- an authentication token is stored in
+  `~/Library/Application\ Support/turborepo/config.json`.
+
+## Inspect the Cache
+
+When turborepo runs a task that has configured `outputs`, it caches those
+outputs, along with the logs from that task in the `node_modules/.cache/turbo/`.
+These artifacts are compressed with `tar`, but you can uncompress and see what's
+in them.
+
+## Build from Source
+
+One of the advantages of JavaScript codebases are that you can open up
+`node_modules/` and edit the code you're running inline. This is not possible
+with `turbo`, because the runnable code is a compiled binary and you cannot edit
+it inline. But because the codebase is Open Source, you can always get
+the source code, modify it, and build it locally. The bulk of this
+documentation is available in the [Contributing Guide][4], but you can use those
+directions even if you aren't planning to make a contribution.
+
+1. Clone the git repo from [`vercel/turbo`][source]
+1. `cd cli`
+1. Make any changes (for example, add more logging)
+1. Run `make`
+1. From _your_ project, use `/:path/:to/:turbo/target/debug/turbo` instead of global
+   turbo or the version of `turbo` installed in your project.
+
+## Common Pitfalls
+
+### The `.turbo` directory
+
+One of the [core concepts][2] behind Turbo is that when a declared input
+changes, the cached outputs for that task are invalidated. As part of running any task,
+Turborepo creates the following directories:
+
+- A `.turbo` at the root of your repo
+- A `.turbo` directory in each workspace if your project is a monorepo (e.g. `apps/my-app/.turbo/`)
+- A `turbo` directory in `node_modules/.cache`
 
-## I'm not seeing any cache hits
+Because the first two directories are not git-ignored by default, you may see an
+issue where you run the same task twice and get a cache missing, even though you
+didn't change anything, because the generated `.turbo` directories are getting included as
+the task _inputs_, and invalidating cache. To avoid this problem, you can limit your
+[`inputs` configuration][r-inputs-config], or add `.turbo/` to your `.gitignore` file. We recommend
+the latter.
```

Comment:
```suggestion
the task _inputs_, and invalidating cache. To avoid this problem, add `.turbo` to your
`.gitignore` file. Alternatively, you can also limit your [`inputs` configuration][r-inputs-config]
so that `.turbo` is not included in the cache inputs.
```